dll/win32/shell32/shlexec.cpp

```diff
@@ -32,6 +32,20 @@ EXTERN_C BOOL PathIsExeW(LPCWSTR lpszPath);
 typedef UINT_PTR (*SHELL_ExecuteW32)(const WCHAR *lpCmd, WCHAR *env, BOOL shWait,
                 const SHELLEXECUTEINFOW *sei, LPSHELLEXECUTEINFOW sei_out);
 
+// Is the current process a rundll32.exe?
+static BOOL SHELL_InRunDllProcess(VOID)
+{
+    WCHAR szModule[MAX_PATH];
+    static INT s_bInDllProcess = -1;
+
+    if (s_bInDllProcess != -1)
+        return s_bInDllProcess;
+
+    s_bInDllProcess = GetModuleFileNameW(NULL, szModule, _countof(szModule)) &&
+                      (StrStrIW(szModule, L"rundll") != NULL);
```

Comment:
Why "rundll" and not "rundll32"? Does ROS even have the 16-bit rundll.exe? And why not skip ahead with PathFindFileName?

Reply:
In the future, we might support 16-bit WoW.

I will use `PathFindFileName`.

Reply:
Fixed.